### DISCUSSION TITLE
Rename storage.objects using SQL = 400 AccessDenied?

### DISCUSSION THREAD

**Discussion by tskytt** [supabase/supabase#3446] [Questions] [answered]
I'm trying to rename storage objects using SQL like this:

`update storage.objects set name = 'avatar2.png' where name = 'avatar1.png' `

But when I try to access the file afterwards I get 400 AccessDenied. Before running the SQL the file is perfectly accessible, but changing the name column this way doesn't work, apparently.

I know I could use Javascript and:

`const { data, error } = await supabase.storage.from('avatars').move('avatar1.png', 'avatar2.png')`

but I want to do it in a SQL function.

Any ideas?

**Comment by inian** [comment, marked as answer]
Hi @tskytt, this is not supported. Changing your database only updates the metadata and does not move the object in S3 behind the scenes. Doing this is problematic since you will also get charged for the object and you cant delete the object via the API (unless you revert the metadata back).

**Reply by tskytt** [reply]
Can I ask another question related to this?

So if I do a `delete from storage.objects where bucket_id =...`, do the actual files in the S3 bucket get deleted?

**Reply by GaryAustin1** [reply]
No S3 file does not get deleted.  This discussion covers what you would have to do to delete files from a database function.  https://github.com/supabase/supabase/discussions/3124#discussion-3561421  You could also use it to rename.
